Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx

```diff
@@ -24,169 +18,62 @@ interface ITestRunDetailsProps {
 const getResultIsWrongAnswerResultType = (run: ITestRunDetailsType) => toLowerCase(run.resultType) !== SubmissionResultType.CorrectAnswer;
 
 const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
-    const { state: { user } } = useAuth();
-    const initialIsCollapsed = testRun.isTrialTest && getResultIsWrongAnswerResultType(testRun);
-    const [ isTestRunDetailCollapsed, setIsTestRunDetailCollapsed ] = useState<boolean>(initialIsCollapsed);
-    const [ isTestInputCollapsed, setIsTestInputCollapsed ] = useState<boolean>(false);
-
-    const testRunHeadingClass = 'testRunHeading';
-    const testRunHeadingClassName = useMemo(
-        () => concatClassNames(
-            styles.testRunHeading,
-            testRunHeadingClass,
-            getResultIsWrongAnswerResultType(testRun)
-                ? styles.wrongTestRunHeading
-                : styles.correctTestRunHeading,
-        ),
-        [ testRun ],
-    );
-
-    const isOutputDiffAvailable = useMemo(
-        () => !isNil(testRun.expectedOutputFragment) && testRun.expectedOutputFragment !== '' &&
-            !isNil(testRun.userOutputFragment) && testRun.userOutputFragment !== '',
-        [ testRun.expectedOutputFragment, testRun.userOutputFragment ],
+    const testRunHeadingClassName = concatClassNames(
+        styles.testRunHeading,
+        getResultIsWrongAnswerResultType(testRun)
+            ? styles.wrongTestRunHeading
+            : styles.correctTestRunHeading,
     );
 
-    const testRunHeadingText = useMemo(() => {
+    const testRunHeadingText = () => {
         const testRunText = `Test #${testRun.orderBy}`;
 
         if (testRun.isTrialTest) {
             return `Zero ${testRunText}`;
         }
 
         return testRunText;
-    }, [ testRun ]);
-
-    const renderResultTypeLabel = useCallback(
-        () => {
-            const result = toLowerCase(testRun.resultType);
-            const type = result === SubmissionResultType.CorrectAnswer
-                ? LabelType.success
-                : result === SubmissionResultType.WrongAnswer
-                    ? LabelType.danger
-                    : LabelType.warning;
-
-            const resultSplit = splitByCapitalLetter(testRun.resultType);
+    };
 
-            return (
-                <Label type={type}>
-                    {resultSplit}
-                </Label>
-            );
-        },
-        [ testRun ],
-    );
-
-    const handleCollapsibleTestInput = useCallback((collapsed: boolean) => {
-        setIsTestInputCollapsed(collapsed);
-    }, []);
+    const handleTestClick = (e: any) => {
+        const { target } = e;
+        const h3Element = target.querySelector('h3');
+        if (!h3Element) { return; }
```

Comment:
since there may be cases where there could not be yet loaded test cases below, which would result in error in the console, since there would not be element to scroll to, this would happen if the users clicks the element before the test cases below are loaded

Reply:
Okay, please separate code in different rows, so its more easily readable.